akka-http-core/src/main/scala/akka/http/impl/engine/http2/BufferedOutletSupport.scala

```diff
@@ -21,78 +21,6 @@ private[http2] trait GenericOutlet[T] {
   def canBePushed: Boolean
 }
 
-/**
- * INTERNAL API
- */
-@InternalApi
-private[http2] class BufferedOutlet[T](outlet: GenericOutlet[T]) extends OutHandler {
-  val buffer: java.util.ArrayDeque[T] = new java.util.ArrayDeque[T]
-  var completed = false
-
-  /**
-   * override to hook into actually pushing, e.g. to keep track how much
-   * has been pushed already (in contract, to being still cached)
-   */
-  protected def doPush(elem: T): Unit = outlet.push(elem)
-
-  outlet.setHandler(this)
-
-  def onPull(): Unit = tryFlush()
-  def push(elem: T): Unit =
-    if (outlet.canBePushed && buffer.isEmpty) doPush(elem)
-    else buffer.addLast(elem)
-
-  def complete(): Unit = {
-    require(!completed, "Can only complete once.")
-    completed = true
-    if (buffer.isEmpty) outlet.complete()
-  }
-
-  def fail(cause: Throwable): Unit = {
-    buffer.clear()
-    outlet.fail(cause)
-  }
-
-  def tryFlush(): Unit = {
-    if (outlet.canBePushed && !buffer.isEmpty)
-      doPush(buffer.pop())
-
-    if (buffer.isEmpty && completed) outlet.complete()
-  }
-}
-
-/**
- * INTERNAL API
- */
-@InternalApi
-private[http2] class BufferedOutletExtended[T](outlet: GenericOutlet[T]) extends OutHandler {
-  case class ElementAndTrigger(element: T, trigger: () => Unit)
-  final val buffer: java.util.ArrayDeque[ElementAndTrigger] = new java.util.ArrayDeque[ElementAndTrigger]
-
-  /**
-   * override to hook into actually pushing, e.g. to keep track how much
-   * has been pushed already (in contract, to being still cached)
-   */
-  protected def doPush(elem: ElementAndTrigger): Unit = {
-    outlet.push(elem.element)
-    elem.trigger()
-  }
-
-  override def onPull(): Unit =
-    if (!buffer.isEmpty) doPush(buffer.pop())
-
-  outlet.setHandler(this)
-
-  final def push(element: T): Unit = pushWithTrigger(element, () => ())
-  final def pushWithTrigger(elem: T, trigger: () => Unit): Unit =
-    if (outlet.canBePushed && buffer.isEmpty) doPush(ElementAndTrigger(elem, trigger))
-    else buffer.addLast(ElementAndTrigger(elem, trigger))
-
-  def tryFlush(): Unit =
-    if (outlet.canBePushed && !buffer.isEmpty)
-      doPush(buffer.pop())
-}
```

Comment:
None of this is used anymore.

Reply:
Would be good to get some input from @jrudolph on why this was used in the first place, seems strange that a custom buffering thing was introduced if emit works just as well.